src/ngraph/op/util/index_reduction.hpp

```diff
@@ -47,6 +47,8 @@ namespace ngraph
                                size_t axis,
                                const element::Type& index_element_type);
 
+                // Additional per-op validation
+                virtual void validate_reduction() const {}
```

Comment:
I'm not sure this needs to be virtual. The `IndexReduction` class is already baking in a lot of assumptions that seem to be tailor-made for `ArgMin` and `ArgMax`. Are we ever going to extend this class with other ops besides those?